### PR TITLE
Reports: Fix inconsistencies with std ones

### PR DESCRIPTION
**Pull request by psiinon**
The traditional html, md and xml reports should be backwards compatible
with the existing core reports. This fixes some of the inconsistencies.

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
WIP until I add some tests

**Comment by psiinon** [comment]
Added a test for the XML report - the others arent generated properly in the unit tests so those are still TODO.

**Comment by psiinon** [comment]
Tests added - ready for review.

**Comment by psiinon** [comment]
Looks like inconsistancies with the HtmlMessage data - I'll try striping out \r \n \t ...

**Comment by psiinon** [comment]
Works ok here but so did the last one ;)

**Comment by psiinon** [comment]
Yey!

**Comment by psiinon** [comment]
Ugh, that merge didnt do what I wanted to :(

**Comment by psiinon** [comment]
Ok, sorted now.

File: addOns/reports/CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,10 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
+
 ### Changed
 - Correct logging of dependency.
+- Inconsistencies between traditional reports and the 'old' core ones
```
[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/reports/CHANGELOG.md:10]: I actually had that in the first PR but then saw that "Correct logging of dependency" was in Changed - should that go in Fixed as well?

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/reports/CHANGELOG.md:10]: Ya probably. Just change the existing heading I guess.

[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/reports/CHANGELOG.md:10]: Done :)

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**